otherlibs/stdlib_alpha/effect.ml

```diff
@@ -1306,53 +1324,55 @@ module Make_rec (Ops : Operations_rec)
       
        We may be able to rewrite this to avoid magic. *)
       let f h a = f (Obj.magic_portable h) a [@nontail] in
-      let k : (a Modes.Portable.t, b, t, unit) continuation = Obj.magic (fiber f) in
+      let k : (a Modes.Portable.t, b, t, unit) continuation = Obj.magic (fiber f : _ @@ unique) in
     (* CR dkalinichenko: here and below, unsafe use of [Obj.magic_portable]:
        the continuation contains mutable data [Mapping.t].
        
        Require [f] to be unique. *)
-      Obj.magic_portable
+    (* CR dkalinichenko: this use of [magic_unique] can be eliminated
+       with mode polymorphism. *)
+      magic_portable_unique
         (Continuation k : (a Modes.Portable.t, (b, unit)
           Result.Portable.Contended.t, unit) Continuation.t)
 
     let fiber (type a b) f =
       let f h a = f (Obj.magic_portable h) a [@nontail] in
-      let k : (a Modes.Portable.t, b, t, unit) continuation = Obj.magic (fiber f) in
-      (Continuation k : (a Modes.Portable.t, (b, unit)
+      let k : (a Modes.Portable.t, b, t, unit) continuation = Obj.magic (fiber f : _ @@ unique) in
+      Obj.magic_unique (Continuation k : (a Modes.Portable.t, (b, unit)
```

Comment:
Better to use `obj_magic_with_unique` instead of `Obj.magic` above so that this `magic_unique` isn't needed.

Reply:
Done.